src/HealthChecks.Aws.Sns/SnsTopicHealthCheck.cs

```diff
@@ -0,0 +1,53 @@
+using Amazon.SimpleNotificationService;
+using Amazon.SimpleNotificationService.Model;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace HealthChecks.Aws.Sns
+{
+    public class SnsTopicHealthCheck : IHealthCheck
```

Comment:
Why did you write 2 health checks instead of 1? At first glance I see no reason why it can not be done as a single health check.

Reply:
From your code I see that there are two entities to check - topic and (optionally) subscriptions to this topic. Both can be handled by a single health check with configured options.

Reply:
So what do you think?

Reply:
It's still single responsibility - check SNS availability. Also remember DRY - now you check topic in two places.

Reply:
Will change. No worries! :-)

Reply:
Reviewed.